pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1080,6 +1082,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	// health check management is coupled with targetPools to prevent leaks. A
+	// target pool is the only thing that requires a health check, so we delete
+	// associated checks on teardown, and ensure checks on setup.
+	if hc != nil {
+		var err error
+		if hc, err = gce.ensureHttpHealthCheck(name, hc.RequestPath, int32(hc.Port)); err != nil || hc == nil {
+			return fmt.Errorf("Failed to ensure health check for %v port %d path %v: %v", name, hc.Port, hc.RequestPath, err)
+		}
+	}
```

Comment:
Yes, I can move it in, but I'll need a second hc != nil check anyway because I want to check the output of ensureHttpHealthCheck (though in theory we should never get err == nil and hc == nil so maybe it's OK)